apps/prairielearn/elements/pl-drawing/mechanicsObjects.js

```diff
@@ -3596,6 +3695,461 @@ mechanicsObjects.byType['pl-controlled-curved-line'] = class extends PLDrawingBa
+      });
+      canvas.add(textObj);
+    }
+
+    return;
+  }
+
+  static get_button_tooltip() {
+    return 'Add capacitor';
+  }
+};
+
+mechanicsObjects.byType['pl-battery'] = class extends PLDrawingBaseElement {
+  static generate(canvas, options) {
+    var gap = options.interval;
+    var theta = Math.atan2(options.y2 - options.y1, options.x2 - options.x1);
+    var d = Math.sqrt(Math.pow(options.y2 - options.y1, 2) + Math.pow(options.x2 - options.x1, 2));
+
+    // Battery "legs"
+    var xm1 = options.x1 + ((d - gap) / 2) * Math.cos(theta);
+    var ym1 = options.y1 + ((d - gap) / 2) * Math.sin(theta);
+    var xm2 = options.x1 + ((d + gap) / 2) * Math.cos(theta);
+    var ym2 = options.y1 + ((d + gap) / 2) * Math.sin(theta);
+
+    let obj1 = new fabric.Line([options.x1, options.y1, xm1, ym1], {
+      stroke: options.stroke,
+      strokeWidth: options.strokeWidth,
+      selectable: false,
+      evented: false,
+      originX: 'center',
+      originY: 'center',
+    });
+    if (!('id' in obj1)) {
+      obj1.id = window.PLDrawingApi.generateID();
+    }
+    let obj2 = new fabric.Line([xm2, ym2, options.x2, options.y2], {
+      stroke: options.stroke,
+      strokeWidth: options.strokeWidth,
+      selectable: false,
+      evented: false,
+      originX: 'center',
+      originY: 'center',
+    });
+    if (!('id' in obj2)) {
+      obj2.id = window.PLDrawingApi.generateID();
+    }
+
+    // Battery lines
+    var cline1 = options.height / 2;
+    var cline2 = options.height;
+
+    var c1x1 = xm1 - cline1 * Math.sin(theta);
+    var c1y1 = ym1 + cline1 * Math.cos(theta);
+    var c1x2 = xm1 + cline1 * Math.sin(theta);
+    var c1y2 = ym1 - cline1 * Math.cos(theta);
+    var c2x1 = xm2 - cline2 * Math.sin(theta);
+    var c2y1 = ym2 + cline2 * Math.cos(theta);
+    var c2x2 = xm2 + cline2 * Math.sin(theta);
+    var c2y2 = ym2 - cline2 * Math.cos(theta);
+
+    let obj3 = new fabric.Line([c1x1, c1y1, c1x2, c1y2], {
+      stroke: options.stroke,
+      strokeWidth: options.strokeWidth,
+      selectable: false,
+      evented: false,
+      originX: 'center',
+      originY: 'center',
+    });
+    if (!('id' in obj3)) {
+      obj3.id = window.PLDrawingApi.generateID();
+    }
+    let obj4 = new fabric.Line([c2x1, c2y1, c2x2, c2y2], {
+      stroke: options.stroke,
+      strokeWidth: options.strokeWidth,
+      selectable: false,
+      evented: false,
+      originX: 'center',
+      originY: 'center',
+    });
+    if (!('id' in obj4)) {
+      obj4.id = window.PLDrawingApi.generateID();
+    }
+
+    canvas.add(obj1, obj2, obj3, obj4);
+
+    if (options.label) {
+      let textObj = new mechanicsObjects.LatexText(options.label, {
+        left: c2x1 + options.offsetx,
+        top: c2y1 + options.offsety + 10,
+        textAlign: 'left',
+        fontSize: options.fontSize,
+        selectable: false,
+        originX: 'center',
+        originY: 'center',
+      });
+      canvas.add(textObj);
+    }
+
+    return;
```

Comment:
There's no need for these empty `return` statements at the end of a function; they're implicit. Ditto for the other new element classes.